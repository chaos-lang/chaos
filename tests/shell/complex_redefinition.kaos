num a = 5
list a = [1]
dict a = {'a': 1}
print a
del a
print a
del a
list a = [1]
num a = 5
bool a = true
str a = 'hey'
print a
