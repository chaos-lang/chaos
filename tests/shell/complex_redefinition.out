Chaos Error:
Module: ~/chaos/__interactive__.kaos
Line: 2
The variable name 'a' is already defined!
Absorbed by Interactive Shell
Chaos Error:
Module: ~/chaos/__interactive__.kaos
Line: 1
The variable name 'a' is already defined!
Absorbed by Interactive Shell
[start] =>
	{id: 0, name: a, 2nd_name: (null), key: (null), scope: main, depth: 0, type: 1, 2nd_type: 0, value_type: 1, role: 0, param_of: } =>
[end]
5
[start] =>
[end]
Chaos Error:
Module: ~/chaos/__interactive__.kaos
Line: 5
Undefined variable: a
Absorbed by Interactive Shell
[start] =>
[end]
Chaos Error:
Module: ~/chaos/__interactive__.kaos
Line: 2
Undefined variable: a
Absorbed by Interactive Shell
Chaos Error:
Module: ~/chaos/__interactive__.kaos
Line: 2
The variable name 'a' is already defined!
Absorbed by Interactive Shell
Chaos Error:
Module: ~/chaos/__interactive__.kaos
Line: 1
The variable name 'a' is already defined!
Absorbed by Interactive Shell
Chaos Error:
Module: ~/chaos/__interactive__.kaos
Line: 1
The variable name 'a' is already defined!
Absorbed by Interactive Shell
[1]
